pkg/v7/configmap/desired.go

```diff
@@ -268,6 +268,113 @@ func (s *Service) checkHelmReleaseExists(ctx context.Context, releaseName string
 	return true, nil
 }
 
+func coreDNSValues(configMapValues ConfigMapValues) ([]byte, error) {
+	calicoCIDRBlock := key.CIDRBlock(configMapValues.CalicoAddress, configMapValues.CalicoPrefixLength)
+	DNSIP, err := key.DNSIP(configMapValues.ClusterIPRange)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	values := CoreDNS{
+		Cluster: CoreDNSCluster{
+			Calico: CoreDNSClusterCalico{
+				CIDR: calicoCIDRBlock,
+			},
+			Kubernetes: CoreDNSClusterKubernetes{
+				API: CoreDNSClusterKubernetesAPI{
+					ClusterIPRange: configMapValues.ClusterIPRange,
+				},
+				DNS: CoreDNSClusterKubernetesDNS{
+					IP: DNSIP,
+				},
+			},
+		},
+		Image: Image{
+			Registry: configMapValues.RegistryDomain,
+		},
+	}
+	json, err := json.Marshal(values)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return json, nil
+}
+
+func defaultValues(configMapValues ConfigMapValues) ([]byte, error) {
+	values := DefaultConfigMap{
+		Image: Image{
+			Registry: configMapValues.RegistryDomain,
+		},
+	}
+	json, err := json.Marshal(values)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return json, nil
+}
+
+func exporterValues(configMapValues ConfigMapValues) ([]byte, error) {
+	values := ExporterValues{
+		Namespace: metav1.NamespaceSystem,
+	}
+	json, err := json.Marshal(values)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return json, nil
+}
+
+func ingressControllerValues(configMapValues ConfigMapValues, releaseExists bool) ([]byte, error) {
+	// controllerServiceEnabled needs to be set separately for the chart
+	// migration logic but is the reverse of migration enabled.
+	controllerServiceEnabled := !configMapValues.IngressControllerMigrationEnabled
+
+	migrationEnabled := configMapValues.IngressControllerMigrationEnabled
+	if migrationEnabled {
+		// Release exists so don't repeat the migration process.
+		if releaseExists {
+			migrationEnabled = false
+		}
+	}
+
+	// tempReplicas is set to 50% of the worker count to ensure all pods can be
+	// scheduled.
+	tempReplicas, err := setIngressControllerTempReplicas(configMapValues.WorkerCount)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	values := IngressController{
+		Controller: IngressControllerController{
+			Replicas: configMapValues.WorkerCount,
+			Service: IngressControllerControllerService{
+				Enabled: controllerServiceEnabled,
+			},
+		},
+		Global: IngressControllerGlobal{
+			Controller: IngressControllerGlobalController{
+				TempReplicas:     tempReplicas,
+				UseProxyProtocol: configMapValues.IngressControllerUseProxyProtocol,
+			},
+			Migration: IngressControllerGlobalMigration{
+				Enabled: migrationEnabled,
+			},
+		},
+		Image: Image{
+			Registry: configMapValues.RegistryDomain,
+		},
+	}
+	json, err := json.Marshal(values)
```

Comment:
Just a brainwave: This is essentially what we do in the `e2etemplates` repo but instead of `json.Marshal` we use the yaml and parse it as a go template. Would it make sense to switch to json there as well? What are your experiences with readability?

Reply:
I think you misunderstood, I saw the benefits of `json` and wanted to ask if it would make sense to switch _to_ json from yaml in our e2e stuff.

Reply:
Ack. Yep I definitely did :)

I like the YAML for humans, JSON for machines pattern. This is what we're using for the per cluster config changes.

For using JSON in e2e I'd be OK with it if we pretty printed the JSON when we need to read it. Reading large amounts of JSON that isn't formatted is horrible IMO.